domain/src/main/java/org/oppia/android/domain/workmanager/WorkManagerConfigurationModule.kt

```diff
@@ -0,0 +1,26 @@
+package org.oppia.android.domain.workmanager
+
+import androidx.work.Configuration
+import androidx.work.DelegatingWorkerFactory
+import dagger.Module
+import dagger.Provides
+import org.oppia.android.domain.oppialogger.loguploader.LogUploadWorkerFactory
+import org.oppia.android.domain.platformparameter.syncup.PlatformParameterSyncUpWorkerFactory
+import javax.inject.Singleton
+
+/** Provides [Configuration] for the work manager. */
+@Module
+class WorkManagerConfigurationModule {
+
+  @Singleton
+  @Provides
+  fun provideWorkManagerConfiguration(
+    logUploadWorkerFactory: LogUploadWorkerFactory,
+    platformParameterSyncUpWorkerFactory: PlatformParameterSyncUpWorkerFactory
+  ): Configuration {
```

Comment:
> https://github.com/oppia/oppia-android/pull/3606#discussion_r686406727

@jcqli @anandwana001 here we have the new `WorkManagerConfigurationModule `which will be injected with `PlatformParameterSyncUpWorkerFactory`. This Factory class requires a will inject the `PlatformParameterSyncUpWorker ` which will directly inject `PlatformParameterService` instance hence we require `NetworkModule ` to be present everywhere the `WorkManagerConfigurationModule `is getting used

Reply:
Ack, I see now. Hm this seems like the correct implementation so I'll defer to @anandwana001 and @rt4914 if they have more perspective on where this could be change. That said, it's being added to an app test so it may not end up being in importing into the app module in prod.

Reply:
Sure that is true